src/meta/service/src/cluster_service.rs

```diff
@@ -176,4 +181,14 @@ impl ClusterService for ClusterServiceImpl {
             nodes: node_list,
         }))
     }
+
+    async fn check_cluster_in_recovery(
+        &self,
+        _request: Request<CheckClusterInRecoveryRequest>,
+    ) -> Result<Response<CheckClusterInRecoveryResponse>, Status> {
+        Ok(Response::new(CheckClusterInRecoveryResponse {
+            status: None,
+            in_recovery: self.barrier_manager.check_status_running().is_err(),
```

Comment:
Can we simply match against `Running`? 🤣 This function will construct an error, capturing the backtrace.